contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,851 @@
+StatefulSet's revision history.
+1. For all the remaining PodTemplates, the controller will sort them in 
+ascending order by the value mapped to their `StatefulSetTemplateRevisionLabel`. 
+This will reconstruct a list of PodTemplates from oldest to newest. Note that, 
+as the revision is monotonically increasing for an individual StatefulSet, and
+as we use ControllerRef to mitigate selector overlap, the StatefulSet's history
+ is a strictly ordered set.
+
+#### History Maintenance 
+In order to prevent the revision history of the StatefulSet from exceeding 
+memory or storage limits, the StatefulSet controller will periodically prune 
+the oldest PodTemplates from the StatefulSet's revision history. 
+
+1. The StatefulSet controller will 
+[reconstruct the revision history](#history-reconstruction) 
+of the StatefulSet.
+1. If the number of PodTemplates in the StatefulSet's revision history is 
+greater than the StatefulSet's `.Spec.RevisionHistoryLimit, the 
+StatefulSet controller will delete PodTemplates, starting with the head of 
+the revision history, until the limit of the revision history is equal to 
+the StatefulSet's `.Spec.RevisionHistoryLimit`.
+1. As a StatefulSet's `.Spec.RevisionHistoryLimit` is always at least two, and 
+as the PodTemplates corresponding to `.Status.TemplateRevision` 
+or `.Status.TargetTemplateRevision` are always the most recent PodTemplates 
+in the revision history, the StatefulSet controller will not delete any 
+`PodTemplates` that represent the current or target revisions.
+
+### Template Updates
+The StatefulSet controller will create PodTemplates upon mutation of the 
+`.Spec.Template` of a StatefulSet.
+
+1. When the StafefulSet controller observes a mutation to a StatefulSet's 
+ `.Spec.Template` it will compare the `.Spec.TemplateRevision` to the 
+ `.Status.TargetTemplateRevision`.
+1. If the `.Spec.TemplateRevision` is equivalent to the 
+`.Status.TargetTemplateRevision`, no update has occurred. Note that, in the 
+event that both are nil, they are considered to be equivalent, and we expect 
+this to occur after an initial upgrade to a version of Kubernetes that supports 
+StatefulSet update form one that does not.
+1. If the `.Status.TemplateRevision` field is nil, and the 
+`.Spec.TemplateRevision` is not nil, then the StatefulSet has no revision 
+history. To initialize its revision history, the StatefulSet controller will 
+set both `.Status.TemplateRevision` and `.Status.TargetTemplateRevision` 
+to `.Spec.TemplateRevision` and 
+[create a new PodTemplate](#podtemplate-creation). 
+1. If the `.Status.TemplateRevision` is not nil, and if the 
+`.Spec.TemplateRevision` is not equal to the `.Status.TargetTemplateRevision`, 
+the StatefulSet controller will do the following.
+    1. The controller will 
+    [reconstruct the revision history](#history-reconsturction) of the 
+    StatefulSet.
+    1. If the revision history of the StatefulSet contains a PodTemplate 
```

Comment:
Why do you do this step instead of always creating a new PodTemplate? It seems weird that `.Spec.TemplateRevision` is not going to equal `.Status.TargetTemplateRevision` if this step triggers.